tests/metrics/test_cuboid_metrics.py

```diff
@@ -8,4 +17,300 @@
         allow_module_level=True,
     )
 
-# TODO(gunnar): Add Cuboid tests!
+CAR_LABEL = "car"
+PEDESTRIAN_LABEL = "pedestrian"
+
+
+def test_cuboid_metrics_simple():
+    # single item, perfect predictions
+    annotations = [
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(0, 0, 0),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(1000, 1000, 1000),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+    ]
+    predictions = [
+        CuboidPrediction(
+            label=CAR_LABEL,
+            position=Point3D(0, 0, 0),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidPrediction(
+            label=CAR_LABEL,
+            position=Point3D(1000, 1000, 1000),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+    ]
+    assert CuboidIOU().eval(annotations, predictions) == ScalarResult(
+        1.0, len(annotations)
+    ), "Unexpected Cuboid IoU result"
+    assert CuboidPrecision().eval(annotations, predictions) == ScalarResult(
+        1.0, len(annotations)
+    ), "Unexpected Cuboid Precision result"
+    assert CuboidRecall().eval(annotations, predictions) == ScalarResult(
+        1.0, len(annotations)
+    ), "Unexpected Cuboid Recall result"
```

Comment:
This doesn't match the calling pattern in the celery service, the filtering is implemented in the base class so this skirts that and introduces a possibility for a mismatch in expectations and behaviour. 

I've actually introduced a bug myself by just testing the `eval` but the base-implemented functionality broke my expectations.
```suggestion
    assert CuboidIOU()(annotations, predictions) == ScalarResult(
        1.0, len(annotations)
    ), "Unexpected Cuboid IoU result"
    assert CuboidPrecision()(annotations, predictions) == ScalarResult(
        1.0, len(annotations)
    ), "Unexpected Cuboid Precision result"
    assert CuboidRecall()(annotations, predictions) == ScalarResult(
        1.0, len(annotations)
    ), "Unexpected Cuboid Recall result"
```

Reply:
Thanks for pointing this out, I just realized that I had overlooked the label matching when I started this PR. I will keep the original implementation (although it is kind of hard to follow for other developers)